src/Umbraco.Web/Mvc/UmbracoControllerFactory.cs

```diff
@@ -13,12 +16,27 @@ namespace Umbraco.Web.Mvc
     public abstract class UmbracoControllerFactory : IFilteredControllerFactory
     {
         private readonly OverridenDefaultControllerFactory _innerFactory = new OverridenDefaultControllerFactory();
+        private readonly IEnumerable<IRenderController> _renderControllers;
+
+        public UmbracoControllerFactory(IEnumerable<IRenderController> renderControllers)
+        {
+            _renderControllers = renderControllers;
+        }
 
         public abstract bool CanHandle(RequestContext request);
 
         public virtual Type GetControllerType(RequestContext requestContext, string controllerName)
         {
-            return _innerFactory.GetControllerType(requestContext, controllerName);
+            var controllerType = _innerFactory.GetControllerType(requestContext, controllerName);
+            if (controllerType == null)
+            {
+                controllerType = _renderControllers
+                    .Select(x => x.GetType())
+                    .FirstOrDefault(x => x.GetMethods(BindingFlags.Public | BindingFlags.Instance)
```

Comment:
If we need to keep this code it will need to be perf optimized before release, we can't have this reflection running on each request.
//cc @nul800sebastiaan + @clausjensen - we can add a task for this to a sprint.